src/Sentry/Http/HttpTransportBase.cs

```diff
@@ -280,31 +280,56 @@ private void ExtractRateLimits(HttpHeaders responseHeaders)
 
         private void HandleSuccess(Envelope envelope)
         {
-            var eventId = envelope.TryGetEventId(_options.DiagnosticLogger);
             if (_options.DiagnosticLogger?.IsEnabled(SentryLevel.Debug) is true)
             {
                 var payload = envelope.SerializeToString(_options.DiagnosticLogger, _clock);
-                _options.LogDebug("Envelope '{0}' sent successfully. Payload:\n{1}", eventId, payload);
+                LogEnvelopeSent(envelope, payload);
             }
             else
             {
-                _options.LogInfo("Envelope '{0}' successfully received by Sentry.", eventId);
+                LogEnvelopeSent(envelope);
             }
         }
 
         private async Task HandleSuccessAsync(Envelope envelope, CancellationToken cancellationToken)
         {
-            var eventId = envelope.TryGetEventId(_options.DiagnosticLogger);
             if (_options.DiagnosticLogger?.IsEnabled(SentryLevel.Debug) is true)
             {
                 var payload = await envelope.SerializeToStringAsync(_options.DiagnosticLogger, _clock, cancellationToken)
                     .ConfigureAwait(false);
 
-                _options.LogDebug("Envelope '{0}' sent successfully. Payload:\n{1}", eventId, payload);
+                LogEnvelopeSent(envelope, payload);
             }
             else
             {
-                _options.LogInfo("Envelope '{0}' successfully received by Sentry.", eventId);
+                LogEnvelopeSent(envelope);
+            }
+        }
+
+        private void LogEnvelopeSent(Envelope envelope, string? payload = null)
+        {
+            var eventId = envelope.TryGetEventId(_options.DiagnosticLogger);
+            if (payload == null)
+            {
+                if (eventId == null)
+                {
+                    _options.LogInfo("Envelope for session successfully received.", eventId);
+                }
+                else
+                {
+                    _options.LogInfo("Envelope '{0}' successfully received.", eventId);
+                }
+            }
+            else
+            {
+                if (eventId == null)
+                {
+                    _options.LogDebug("Envelope for session successfully received. Content:\n{1}", eventId, payload);
```

Comment:
The `\n` will be replaced with ` ` due to #1756.  Might as well not put them.

Also, I believe they should all say `sent`, not `received`.

Reply:
And I'm not entirely sure that `eventId` being absent *only* happens with sessions.  We should just omit `for session`.  No need to check.

Reply:
> being absent only happens with sessions

i did wonder about that. 

> We should just omit for session

done

Reply:
> I believe they should all say sent, not received.

this comes from the `HandleResponse`. so received seems more correct?? (happy to change it)